### ISSUE TITLE
Algolia search depends on `typing` which is in Python3.7 standard library

### ISSUE THREAD

**Issue by fernando-mc** [algolia/algoliasearch-client-python#422]
- Algolia Client Version: 2.0.1
- Language Version: Python 3.7

### Description
When using the algoliasearch library in a python3.7 environment it installs `typing` which causes some sort of conflict with the standard library. 

Because the library is only required with version of Python 3.5 and below can you add that condition to the requirements file?

See this for context on the issue - https://github.com/python/typing/issues/573

This is a solution implemented elsewhere:
https://github.com/MartinoMensio/allennlp/commit/a7d7b8fd2ff1725f546372850d37b7ab43468465

### Steps To Reproduce
In my case this was actually frustrating me. I can reproduce this issue within an AWS python v3.7.3 runtime but not in my local python 3.7.3 runtime.

My steps to reproduce this involved:
Deploying an application with the algoliasearch dependency to the AWS Python 3.7 runtime.
Noting the stack traces pointing to the Algolia library and the typing library:

File "/var/task/typing.py", line 1004, in __new__
self._abc_registry = extra._abc_registry
AttributeError: type object 'Callable' has no attribute '_abc_registry'

I can open a PR for this fix if you think it would be fixed. At the moment I have to run some custom logic not to deploy the typing library with Algolia search.

**Comment by nunomaduro** [comment]
I would appreciate a pull request to better understand the issue, thanks!

**Comment by fernando-mc** [comment]
I'll put in the PR shortly. The issue can be reproduced with the following steps which roughly equate to how packages are handled in the AWS Lambda Python3.7 runtime.

This assumes a Python 3.7.x version:
1. Install packages in the current directory with a command like `pip3 install algoliasearch -t .`
2. Then enter the Python3.7 interpreter `python3`
3. Run `from algoliasearch.search_client import SearchClient`

Then you will see the following stack trace:

```Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/fernando/Desktop/test/algoliasearch/search_client.py", line 5, in <module>
    from typing import Optional, Union, List
  File "/Users/fernando/Desktop/test/typing.py", line 1356, in <module>
    class Callable(extra=collections_abc.Callable, metaclass=CallableMeta):
  File "/Users/fernando/Desktop/test/typing.py", line 1004, in __new__
    self._abc_registry = extra._abc_registry
AttributeError: type object 'Callable' has no attribute '_abc_registry'```

**Comment by nunomaduro** [comment]
This is was been fixed on the version `v2.0.2`.
